### PR TITLE
Fix multiple calls to tree display in multi-widget deletion.

### LINKED ISSUE #1351: deleting many widgets at once is slow
when you use the multiselect and then delete the selected widgets, it is waaay slower than deleting them one by one.

### PR DESCRIPTION
**Pull request by rogerl50**
Fixes #1351.

`jeDisplayTree` wass being called by `removeWidgetLocal` for each widget in the selection. This PR wraps the  for loop in `je_removeWidget` with `batchStart` and `batchEnd`.

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: https://test.virtualtabletop.io/PR-1367/pr-test (or any other room on that server)

After merging, a backup will be available at `https://beta.virtualtabletop.io/editor/PR1367-pr-test`.

**Comment by rogerl50** [comment]
@Ingegneus Your name doesn't appear to show up in the list of possible reviewers. But please have a look.

**Comment by 96LawDawg** [comment]
I tested it on the SmashUp game mentioned in #1351.  It is very much faster in this PR.  Multi-select and delete is still a bit slower than average in that game, but it is a huge file with lots of widgets and that is probably to be expected.  But the delay is fine, in my opinion, after this PR.

**Comment by Ingegneus** [comment]
I measured the time it takes to delete 10 cards from a hand. on the unoptimized it took 5 min 20 s and on the optimized one 36 s. I guess this is in line with what one would expect. 10 times faster for 10 widgets.

**Comment by ArnoldSmith86** [comment]
> 36 s

It takes 36s to rebuild the widget tree for you?

**Comment by Ingegneus** [comment]
yes 🙃

**Comment by ArnoldSmith86** [comment]
How about now?

**Comment by Ingegneus** [comment]
oh wow ok now it is almost instantaneous

**Review by ArnoldSmith86 — APPROVED** [review]
@rogerl50 `jeSelectedIDs` was very slow and called in each iteration of `jeDisplayTreeAddWidgets`. I changed that.

I also noticed that the tree was rebuilt twice because `jeEmpty` did it _and_ `batchEnd` did it. After looking at the code, I'm pretty sure that every function that calls `jeEmpty` updates the tree afterwards anyway so I removed it from `jeEmpty`.

You get an approval from me for your changes. Please take another look at my changes...

**Review by 96LawDawg — APPROVED** [review]
Approved based on testing.  My times went from (an estimated ... not measured) 30s on production to 6-7 seconds after the first iteration to instant after the latest changes.

**Pull request merged**
